src/client.rs

```diff
@@ -995,7 +995,8 @@ impl<T: Read + Write> Session<T> {
     ///
     /// See [`extensions::idle::Handle`] for details.
     pub fn idle(&mut self) -> Result<extensions::idle::Handle<T>> {
-        extensions::idle::Handle::make(self)
+        let sender = self.unsolicited_responses_tx.clone();
```

Comment:
Re-using the `unsolicited_responses_tx` for this means that you'll also get any unsolicited messages from _before_ the IDLE command was issued. Do you think that may be problematic?

Reply:
Well, that way unsolicited responses that are sent between the previous command and the idle command aren't lost. And with the current implementation, the user should empty the unsolicited response queue after each command or it will grow indefinitely. Or maybe the library should clear the queue at the start of each command if the user has not done so.

Reply:
I'm actually not sure what's best here. Forcing a clearing of the queue at every command seems like something the user might not be expecting. Maybe it'd make sense to purge it before you enter IDLE mode though?

Reply:
Maybe. But currently if the user does nothing about unsolicited responses, the queue will keep growing forever.

Reply:
Hmm, yeah, I honestly don't know what to do here. Maybe we make the channel a bounded channel instead, and drop unsolicited responses if it's full? And then clear before entering IDLE.

Reply:
What about this:
- we add a request_unsolicited(category_flags) method to Session
- handle_unsolicited() enqueues only responses that are in the requested categories
- the category flags could default to be empty, so that users not explicitly asking for unsolicited responses wont waste memory with an infinitely growing queue, and we can assume that the user does handle responses he asks for.
- document the idle iterator as returning the not yet handled responses first, if any.
Alternatively, the default category_flags could be STATUS|RECENT|EXISTS|EXPUNGE if you prefer compatibility with the current behavior.

Reply:
I like that plan! "Muting" all by default seems reasonable. I think finding a good name for the "unmasking" method is important, but beyond that this seems like a good strategy.